upgrade.sh

```diff
@@ -52,6 +52,66 @@ function unbound_psp_check() {
     echo "cray-unbound-coredns-psp check Done"
 }
 
+# Function to check cray-sysmgmt-health chart with app version 9.3.1 for prometheus-operator and retain old PVs data.
+function sysmgmt_health() {
+    echo "Checking for chart version of cray-sysmgmt-health"
+    version="45.1"
+    if [ $(helm ls -o json --namespace sysmgmt-health| jq -r --argjson version $version '.[] | select(.app_version | sub(".[0-9]$";"") | tonumber | . < $version).name') ]
```

Comment:
The test block and ```$(``` are unnecessary here, if you're just looking for rc=0 from a command if command; then... is the simplest form no need to invoke ```$(``` or test in those cases.

Reply:
We are not checking for rc=0 from the above command rather we are checking if the cray-sysmgmt-health chart version is < 45.1.1 then execute the further steps.

Reply:
Added [ ! -z "$(command)" ]  to check for null condition.

Reply:
Cool thanks makes it much more clear what condition we're actually testing in the if test block.